crypto/rsa/rsa_pss.c

```diff
@@ -191,6 +201,8 @@ int RSA_padding_add_PKCS1_PSS_mgf1(RSA *rsa, unsigned char *EM,
         if (RAND_bytes(salt, sLen) <= 0)
             goto err;
     }
+    if (hLen < 1 || hLen - 1 > emLen)
```

Comment:
The next line is ```emLen - hLen - 1```.

So let's say that ```hLen``` is 0. Subtracting ```1``` from it makes it negative. This has to be avoided. This explains the first condition.

As for the second condition, let's say ```hLen - 1``` resolves to a value larger than ```emLen```. Then the computation ```emLen - hLen - 1``` yields a negative value. Perhaps I should change the ```>``` in the second condition to ```>=```.

Note: I didn't check whether any of my checks can be true in any case. For example I didn't check if ```hLen``` can actually be 0 at this point. I implemented the checks so that there can be no doubt over over/underflows.